indra/explanation/pathfinding/util.py

```diff
@@ -76,45 +83,59 @@ def signed_nodes_to_signed_edge(source, target):
         return None, None, None
 
 
-def get_sorted_neighbors(G, node, reverse, force_edges=None):
-    """Sort the returned neighbors in descending order by belief
+def get_sorted_neighbors(
+        G: nx.DiGraph,
+        node: Node,
+        reverse: bool,
+        force_edges: Optional[List[Edge]] = None,
+        edge_filter: Optional[Callable[[nx.DiGraph, Node, Node], bool]] = None
+) -> List[Node]:
+    """Filter and sort neighbors of a node in descending order by belief
 
     Parameters
     ----------
     G : nx.DiGraph
         A networkx DiGraph
-    node : str|int
-        A valid networkx node name
+    node : Node
+        A valid node name or signed node name
     reverse : bool
         Indicates direction of search. Neighbors are either successors
         (downstream search) or predecessors (reverse search).
-    force_edges : list
+    force_edges : Optional[List[Tuple[Node, Node]]]
         A list of allowed edges. If provided, only allow neighbors that
         can be reached by the allowed edges.
+    edge_filter : Optional[Callable[[nx.DiGraph, Node, Node], bool]]
+        If provided, must be a function that takes three arguments: a graph
+        g, and the nodes u, v of the edge between u and v. The function must
+        return a boolean. The function must return True if the edge is
+        allowed, otherwise False.
+
+    Returns
+    -------
+    List[Node]
+        A list of nodes representing the filtered and sorted neighbors
     """
+    if force_edges:
+        neigh_edges = G.in_edges if reverse else G.out_edges
+        ix = 0 if reverse else 1
+        neighbors = (e[ix] for e in
+                     set(neigh_edges(node)).intersection(set(force_edges)))
```

Comment:
you might be able to get away with `set.intersection(neigh_edges(node), force_edges`